mermithid/misc/FakeTritiumDataFunctions.py

```diff
@@ -0,0 +1,655 @@
+    convolved = convolve(beta_rates, lineshape_rates, mode='same')
+    below_Kmin = np.where(K < Kmin)
+    np.put(convolved, below_Kmin, np.zeros(len(below_Kmin)))
+    return convolved
+
+
+
+#Convolution of background and lineshape using scipy.signal.convolve
+def convolved_bkgd_rate_arrays(K, Kmin, Kmax, lineshape, ls_params, min_energy, max_energy):
+    """K is an array-like object
+    """
+    energy_half_range = max(max_energy, abs(min_energy))
+
+    dE = K[1] - K[0]
+    n_dE_pos = round(energy_half_range/dE) #Number of steps for the lineshape for energies > 0
+    n_dE_neg = round(energy_half_range/dE) #Same, for energies < 0
+    K_lineshape =  np.arange(-n_dE_neg*dE, n_dE_pos*dE, dE)
+
+    #Generating finely spaced points on the lineshape
+    if lineshape=='gaussian':
+        lineshape_rates = gaussian(K_lineshape, [ls_params[0], 0])
+    elif lineshape=='simplified_scattering' or lineshape=='simplified':
+        lineshape_rates = simplified_ls(K_lineshape, 0, ls_params[0], ls_params[1], ls_params[2], ls_params[3], ls_params[4], ls_params[5])
+    elif lineshape=='detailed_scattering' or lineshape=='detailed':
+        lineshape_rates = spectrum_func(K_lineshape/1000., ls_params[0], 0, ls_params[1], 1)
+
+    bkgd_rates = np.full(len(K), bkgd_rate())
+    if len(K) < len(K_lineshape):
+        raise Exception("lineshape array is longer than Koptions")
+
+    #Convolving
+    convolved = convolve(bkgd_rates, lineshape_rates, mode='same')
+    below_Kmin = np.where(K < Kmin)
+    np.put(convolved, below_Kmin, np.zeros(len(below_Kmin)))
+    return convolved
+
+
+
+"""
+Functions to calculate number of events to generate
+"""
+##Fraction of events near the endpoint
+##Currently, this only holds for the last 13.6 eV of the spectrum
+#def frac_near_endpt(Kmin, Q, mass, atom_or_mol='atom'):
+#    A = integrate.quad(spectral_rate, Kmin, Q-mass, args=(Q,mass))
+#    B = integrate.quad(spectral_rate, V0, Q-mass, args=(Q,mass)) #Minimum at V0 because electrons with energy below screening barrier do not escape
+#    f = (A[0])/(B[0])
+#    if atom_or_mol=='atom':
+#        return 0.7006*f
+#    elif atom_or_mol=='mol' or atom_or_mol=='molecule':
+#        return 0.57412*f
+#    else:
+#        print("Choose 'atom' or 'mol'.")
+
+
+#Convert [number of particles]=(density*volume*efficiency) to a signal activity A_s, measured in events/second.
+def find_signal_activity(Nparticles, m, Q, Kmin, atom_or_mol='atom', nTperMolecule=2):
+    br = frac_near_endpt(Kmin, Q, m, atom_or_mol)
+    Thalflife = 3.8789*10**8
+    A_s = Nparticles*np.log(2)/(Thalflife)*br
+    if atom_or_mol=='atom':
+        return A_s
+    elif atom_or_mol=='mol' or atom_or_mol=='molecule':
+        #Multiply molecular activity by the average number of tritium atoms per molecule.
+        #For example: A gas of half HT and half T2 would have nTperMolecule==1.5.
+        return nTperMolecule*A_s
+
+"""
+Function to calculate efficiency
+"""
+
+def Frequency(E, B):
```

Comment:
This function is a more generic one and is duplicated in https://github.com/project8/mermithid/blob/master/mermithid/processors/misc/FrequencyEnergyConversionProcessor.py#L24
If we decide on keeping it here, then we should remove the one in the FrequencyEnergyConversion processor and use this one

Reply:
It is not duplicated. This is energy to frequency conversion. The one in FrequencyEnergyConversionProcessor converges from frequency to energy. I still moved it a new place: mist/ConversionFunctions.